cypress/e2e/downloads.cy.ts

```diff
@@ -1,354 +1,309 @@
-        expect(requestUrl.indexOf('flavorId')).to.not.equal(-1);
+  });
+
+  describe('image entry', () => {
+    beforeEach(() => {
+      cy.intercept('GET', '**/fixture/ocean.jpeg/**', {fixture: 'ocean.jpeg'});
+    });
+
+    it('should make HEAD for the image download url', done => {
+      mockKalturaBe('base-entry.json', 'kaltura-assets-image-entry.json');
+      const imageSourcesConfig = {
+        id: '1234',
+        downloadUrl: 'https://cfvod.kaltura.com/p/3188353/sp/318835300/raw/entry_id/1_vznuyyho/version/100001',
+        image: [
+          {
+            mimetype: 'image/jpeg',
+            url: './fixture/ocean.jpeg'
+          }
+        ]
+      };
+
+      let requestCount = 0;
+      cy.intercept('**/raw/**', () => {
+        ++requestCount;
       });
-      it('should not contain a flavor id if flavor id is not set', async () => {
-        await loadPlayerAndSetMedia();
-        expect(requestUrl.indexOf('flavorId')).to.equal(-1);
+
+      loadPlayer().then(player => {
+        setMedia(player, {ks: '5678'}, imageSourcesConfig);
+        cy.get('[data-testid="download-overlay-button"]')
+          .should('exist')
+          .then(() => {
+            expect(requestCount).to.equal(1);
+            done();
+          });
       });
     });
-    describe('flavor param id', () => {
-      it('should use flavor param id if it is set', async () => {
-        await loadPlayerAndSetMedia({plugins: {uiManagers: {}, download: {flavorParamId: '1234'}}});
-        expect(requestUrl.indexOf('flavorParamId/1234')).not.to.equal(-1);
+    it('should fallback to thumbnail service request', done => {
+      mockKalturaBe('base-entry.json', 'kaltura-assets-image-entry.json');
+      const imageSourcesConfig = {
+        id: '1234',
+        image: [
+          {
+            mimetype: 'image/jpeg',
+            url: './fixture/ocean.jpeg'
+          }
+        ]
+      };
+
+      let requestCount = 0;
+      cy.intercept('**/ks/5678*', () => {
+        ++requestCount;
       });
-      it('should use default flavor param id if it is not set', async () => {
-        await loadPlayerAndSetMedia();
-        expect(requestUrl.indexOf('flavorParamId/0')).not.to.equal(-1);
+
+      loadPlayer().then(player => {
+        setMedia(player, {ks: '5678'}, imageSourcesConfig);
+        cy.get('[data-testid="download-overlay-button"]')
+          .should('exist')
+          .then(() => {
+            expect(requestCount).to.equal(1);
+            done();
+          });
       });
     });
   });
 
-  describe('download overlay button', () => {
-    describe('if content type is not video', () => {
-      beforeEach(() => {
-        cy.intercept('**/playManifest/**', {
-          headers: {
-            'content-type': 'image/gif'
-          },
-          fixture: 'video.mp4'
+  describe('download overlay', () => {
+    describe('download overlay configuration', () => {
+      it('should display all kinds of assets', () => {
```

Comment:
wdym by "all kinds" ? all the different types of assets at once ?

Reply:
yes, showing flavors, captions and attachments.